src/aktualizr_secondary/secondary_tcp_server.cc

```diff
@@ -30,50 +35,68 @@ void SecondaryTcpServer::run() {
   }
   LOG_INFO << "Secondary TCP server listening on " << listen_socket_.toString();
 
+  {
+    std::unique_lock<std::mutex> lock(running_condition_mutex_);
+    is_running_ = true;
+    running_condition_.notify_all();
+  }
+
   while (keep_running_.load()) {
     sockaddr_storage peer_sa{};
     socklen_t peer_sa_size = sizeof(sockaddr_storage);
 
     LOG_DEBUG << "Waiting for connection from Primary...";
     int con_fd = accept(*listen_socket_, reinterpret_cast<sockaddr *>(&peer_sa), &peer_sa_size);
     if (con_fd == -1) {
+      // why are we aborting here?
```

Comment:
What would you do instead?

Reply:
Just continue accepting/listening for a new connection, otherwise Secondary becomes unavailable. On the other hand, the secondary daemon is managed by systemd which is supposed to restart it so it starts listening for a new connection again.

Reply:
> t I'd prefer to remove the comment to prevent future confusion, unless you either want to make it an actual explanation of why it is aborting or a TODO with a specific improvement proposal.

I'll add an explanation and leave the `break` and let systemd do the work here.